db/db.go

```diff
@@ -393,6 +393,13 @@ func configureClient(opts options.ToolOptions) (*mongo.Client, error) {
 			AuthSource:    opts.GetAuthenticationDatabase(),
 			AuthMechanism: opts.Auth.Mechanism,
 		}
+		if cs.AuthMechanism ==  "MONGODB-AWS" {
+			cred.Username = cs.Username
+			cred.Password = cs.Password
+			cred.AuthSource = cs.AuthSource
+			cred.AuthMechanism = cs.AuthMechanism
+			cred.AuthMechanismProperties = cs.AuthMechanismProperties
+		}
```

Comment:
I tried doing what the technical design suggests -- to simplify the logic by just using `cs` instead of `opts` -- but while that worked for AWS auth, it doesn't work for any other auth mechanism.

Reply:
Why didn't it work?

Reply:
The values must be slightly different since I kept getting `'(Unauthorized) command insert requires authentication'` for the non-AWS tests, e.g. [this patch](https://evergreen.mongodb.com/version/5ee004440305b9741fa8b50e).

Reply:
Okay, pity it didn't just work. I'll open a ticket for the refactor since it's probably more involved. But what you have here is fine as it is.